File: medium/807. Max Increase to Keep City Skyline.java
class Solution {
    public int maxIncreaseKeepingSkyline(int[][] grid) {
        int n = grid.length, result = 0;
        int[] rMax = new int[n]; 
        int[] cMax = new int[n]; 
        
        for(int i = 0; i < n; i++)
            for(int j = 0; j < n; j++)
            {
                rMax[i] = Math.max(rMax[i], grid[i][j]);
                cMax[j] = Math.max(cMax[j], grid[i][j]);
            }
        
        
        for(int i = 0; i < n; i++)
            for(int j = 0; j < n; j++)
            {
                result += Math.min(rMax[i], cMax[j]) - grid[i][j];
            }
        
        return result;
    }
}

File: medium/807. Max Increase to Keep City Skyline.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int maxIncreaseKeepingSkyline(vector<vector<int>>& grid) {
        vector<int> cMax, rMax;
        int n = grid.size();
        int result = 0;
        for(int i = 0; i<n; i++)
            for(int j=0; j<n; j++)
            {
                cMax.push_back(max(cMax[j], grid[i][j]));
                rMax.push_back(max(rMax[i], grid[i][j]));
            }


        for(int i = 0; i<n; i++)
            for(int j=0; j<n; j++)
            {
                result += min(cMax[j], rMax[i]) - grid[i][j];
            }

        return result;
    }
};